### PR TITLE
General code improvements

### PR DESCRIPTION
**Pull request by asn-d6**
- Better memory management in compute_challenges()
- Forgotten retval check in compute_aggregated_poly_...()
- General niceties around the code

### CODE REVIEW

**Comment by xrchz** [comment]
LGTM. Is there more coming?

**Comment by asn-d6** [comment]
No new stuff for now. Let's merge this!

File: src/c_kzg_4844.c [hunk 1]
```diff
@@ -1172,10 +1184,11 @@ static C_KZG_RET compute_aggregated_poly_and_commitment(Polynomial *poly_out, g1
     poly_lincomb(poly_out, polys, r_powers, n);
 
     ret = g1_lincomb(comm_out, kzg_commitments, r_powers, n);
+    if (ret != C_KZG_OK) goto out;
 
 out:
     free(r_powers);
-    return C_KZG_OK;
+    return ret;
```
Comment (jtraglia) [inline src/c_kzg_4844.c:1191]: Good catch! That was probably my fault.

**Review by jtraglia — APPROVED** [review]
LGTM 👍 I like these changes.

**Pull request merged**